apps/managedidentity/managedidentity.go

```diff
@@ -105,6 +140,11 @@ func WithHTTPClient(httpClient ops.HTTPClient) ClientOption {
 //
 // Options: [WithHTTPClient]
 func New(id ID, options ...ClientOption) (Client, error) {
+	source, err := GetSource(id)
```

Comment:
What is the scope of this source? Is it for the app level or request level?

Reply:
It is the equivalent of the .NET implementation:

```
        public ManagedIdentityClient(RequestContext requestContext)
        {
            using (requestContext.Logger.LogMethodDuration())
            {
                _identitySource = SelectManagedIdentitySource(requestContext);
            }
        }
```

For getting it when we create the ManagedIdentityClient